contrib/babelfishpg_tsql/src/fts_parser.y

```diff
@@ -195,6 +205,93 @@ static char
+ * Case 6: '"word1 word2* "' && '" word1 word2* "' are treated as simple terms over SQL server
+ * Trivial Case: spaces before and after double quotes, Example - '   "word1 word2" ' = 'word1<->word2'
+ */
+static char
+*translate_prefix_term(const char* inputStr) {
+    char                *outputStr;
+    char                *trimmedOutputStr;
+    StringInfoData      output;
+    static const char   *specialChars = "~!&|@#$%^+=\\;:<>?.\\/`'_";
+
+
+    /* Check for empty input - this should not be possible based on lexer rules, but check just in case */
+    if (!inputStr || !*inputStr) {
+      ereport(ERROR,
+          (errcode(ERRCODE_INTERNAL_ERROR),
+           errmsg("Null or empty full-text predicate.")));
+    }
+    
+    outputStr = pstrdup(inputStr);
+
+    /* removing leading and trailing spaces outside of double quotes */
+    trim(outputStr, false);
+  
+    /* 
+     * removing leading spaces, for the phrase enclosed in double quotes
+     * search string with trailing spaces are identified as simple terms by the lexer
+     */
+    trim(outputStr, true);
+    
+    initStringInfo(&output);
+    trimmedOutputStr = outputStr;
+
+    /* 
+     * Rewriting search string in format word1:*<->word2:*  
+     */
+    while(*trimmedOutputStr) {
+        if (strchr(specialChars, *trimmedOutputStr) != NULL) {
+            ereport(ERROR,
+                (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                errmsg("Special characters in the prefix term search condition are not currently supported in Babelfish")));
+        }
+        if (*trimmedOutputStr == ' ' || *trimmedOutputStr == '*') {
+            /* 
+             * Removing multiple spaces and * from the search string 
+             * If a space is encountered, we remove all the next occurances of * and spaces 
+             * before end of the input or if next word is encountered
+             * Case 1: '"word1   * * ** *"' = 'word1:*'
+             * Case 2: '"word1   * * ** * word2*"' = 'word1:*<->word2:*'
+             * Case 3: '"word1* * ** *"' = 'word1:*'
+             * Case 4: '"word1* * *      *** * word2*"' = 'word1:*<->word2:*'
+             */
+            while(*(trimmedOutputStr + 1) && (*(trimmedOutputStr + 1) == '*' || *(trimmedOutputStr + 1) == ' ')) {
+                trimmedOutputStr++;
+            }
+            if(output.len > 0){
+              if(*(trimmedOutputStr + 1) == '\0') {
+                    appendStringInfoString(&output, ":*");
+              }
+              else {
+                    appendStringInfoString(&output, ":*<->");
+              }
+            }
+        } else if (*trimmedOutputStr == '\n' || *trimmedOutputStr == '\t') {
+                /*
+                 * To match the behavior of SQL server.
+                 * When FTS is performed over prefix_term,
+                 * it outputs no result
+                 */ 
+                pfree(outputStr);
+                output.data = "";
```

Comment:
setting `output.data` to a string literal after initialization without proper cleanup will cause memory leaks. Use `resetStringInfo(&output)` instead to properly clean up the previously allocated memory.